acme/challenge.go

```diff
@@ -411,7 +418,12 @@ func deviceAttest01Validate(ctx context.Context, ch *Challenge, db DB, jwk *jose
 		//
 		// Note: We might want to use an external service for this.
 		if data.UDID != ch.Value && data.SerialNumber != ch.Value {
-			return storeError(ctx, db, ch, true, NewError(ErrorBadAttestationStatementType, "permanent identifier does not match"))
+			subproblem := NewSubproblemWithIdentifier(
+				ErrorRejectedIdentifierType,
+				Identifier{Type: "permanent-identifier", Value: ch.Value},
+				"challenge identifier %q doesn't match any of the attested hardware identifiers %s", ch.Value, []string{data.UDID, data.SerialNumber},
+			)
+			return storeError(ctx, db, ch, true, NewError(ErrorBadAttestationStatementType, "permanent identifier does not match").WithAdditionalErrorDetail().AddSubproblems(subproblem))
```

Comment:
Instead of `WithAddinionalErrorDetails()` why not a new constructor like `NewDetailedError()`

Reply:
It just was my first idea and did fit in with the already existing fluent pattern for `AddSubproblems`. It also works for `Errors` from both `NewError` and `NewErrorISE`, although the additional detail is explicitly blocked for the latter, as a bit of an additional safety net.

Reply:
`AddSubproblems` make more sense because the NewError has a fmt-like interface.